HHS/Head-Start-TTADP
# CD: Add env variables to application

**Description of change**
- Pass env variables to application in deployment pipeline
- Add sandbox: I also added back in the sandbox deployment configuration. I think this could be useful as a way for developers to test changes during feature development. I've found it critical for my work. I added a short explanation of the sandbox environment to the Deployment section of the README.md.
- Give each deployment a unique application name (constructed with `tta-smarthub-((env))` in manifest.yml). Cloud.gov uses the application name to construct the route that serves the application. Only one application can be served at each route, thus necessitating unique application name.

**NOTE: Before this code is merged to the upstream HHS/Head-Start-TTADP repo's main branch, the following variables must be created in circle ci:**
STAGING_AUTH_CLIENT_ID
STAGING_AUTH_CLIENT_SECRET
STAGING_SESSION_SECRET

**How to test**
- Before merge, check on dev deployment. Dev application is [deployed](https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP?branch=main), cloud.gov [instance is healthy](https://dashboard.fr.cloud.gov/applications/2oBn9LBurIXUNpfmtZCQTCHnxUM/ce39fffc-0a4f-4b06-a202-4ab147effb64/summary), and application is served at https://tta-smarthub-dev.app.cloud.gov.  
- After merge, check new staging deployment by looking at circle ci build, cloud.gov instance, and url https://tta-smarthub-staging.app.cloud.gov.

**Issue(s)**
* closes https://github.com/HHS/Head-Start-TTADP/issues/47

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated - _updated README, added comments_
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
